contribs/src/main/java/com/netflix/conductor/contribs/http/HttpTask.java

```diff
@@ -69,9 +72,12 @@ public void start(Workflow workflow, Task task, WorkflowExecutor executor) throw
 			task.setReasonForIncompletion(MISSING_REQUEST);
 			task.setStatus(Status.FAILED);
 			return;
-		} else {
-			if (input.getServiceDiscoveryQuery() != null) {
-				url = lookup(input.getServiceDiscoveryQuery());
+		} else if (input.getServiceDiscoveryQuery() != null) {
+			url = lookup(input.getServiceDiscoveryQuery());
+
+			if (null == url) {
+				logger.error("SRV lookup of: " + input.getServiceDiscoveryQuery() + " failed, falling back to URI: " + StringUtils.defaultIfEmpty(input.getUri(), "[UNDEFINED]"));
+				// Oleksiy: Should the code return from this case or is it OK to continue?
```

Comment:
We should fail the task as SD failed and does not make sense to proceed.

Reply:
OK. I'll update this and commit the changes this morning.

Reply:
Throw something like this ... "Service discovery failed for " + input.getServiceDiscoveryQuery() + ". No records found" ... or up to you.

Reply:
Fixed in 2891b7ae.